src/core/sys/posix/pthread.d

```diff
@@ -499,8 +499,11 @@ int pthread_key_delete(pthread_key_t);
 int pthread_mutex_destroy(pthread_mutex_t*);
 int pthread_mutex_init(pthread_mutex_t*, pthread_mutexattr_t*) @trusted;
 int pthread_mutex_lock(pthread_mutex_t*);
+int pthread_mutex_lock(shared(pthread_mutex_t)*);
```

Comment:
> It's also pretty weird to have shared on only some of them when they're all clearly in the same boat.

Not all of them can be called concurrently on the same object. For example `pthread_mutex_init` and `pthread_mutex_destroy` are not thread-safe - they should be called only when the (conceptual) ref count of the mutex drops to 1. Also removing the non-`shared` overloads would be breaking change.

Functions like `pthread_rwlock_*lock` should also have shared overloads, but I prefer to leave this for another day.